src/query/phrase_query/phrase_weight.rs

```diff
@@ -50,27 +50,14 @@ impl PhraseWeight {
             .map(|similarity_weight| similarity_weight.boost_by(boost));
         let fieldnorm_reader = self.fieldnorm_reader(reader)?;
         let mut term_postings_list = Vec::new();
-        if reader.has_deletes() {
```

Comment:
I think the point of this was to avoid paying for the position check if we have deletes.

Reply:
I saw that `read_postings` and `read_postings_no_deletes` is the same. The bitset is passed as parameter, so this seems to be an outdated API. I'll remove it in a follow-up.


```rust
    pub fn read_postings(
        &self,
        term: &Term,
        option: IndexRecordOption,
    ) -> io::Result<Option<SegmentPostings>> {
        self.get_term_info(term)?
            .map(move |term_info| self.read_postings_from_terminfo(&term_info, option))
            .transpose()
    }

    pub(crate) fn read_postings_no_deletes(
        &self,
        term: &Term,
        option: IndexRecordOption,
    ) -> io::Result<Option<SegmentPostings>> {
        self.get_term_info(term)?
            .map(|term_info| self.read_postings_from_terminfo(&term_info, option))
            .transpose()
    }
```

Reply:
can we open a ticket to clean this up?

Reply:
https://github.com/quickwit-oss/tantivy/issues/2525